src/redis-check-aof.c

```diff
@@ -224,6 +245,14 @@ int redis_check_aof_main(int argc, char **argv) {
 
     off_t pos = process(fp);
     off_t diff = size-pos;
+
+    /* In truncate-to-timestamp mode, just exit if there is nothing to truncate. */
+    if (diff == 0 && to_timestamp) {
+        printf("Truncate nothing in AOF to timestamp %ld\n", to_timestamp);
+        fclose(fp);
+        exit(0);
+    }
```

Comment:
so we skip the report (blow), if we didn't do any truncation?
i.e the file was valid and timestamp was not found.

and if there was a truncation, this can be only due to timestamp, since we don't allow mixing it with `--fix`?

i think there's room to support both side by side, and always print the detailed report.
am i missing anything?

Reply:
@oranagra Sorry for delayed reply.
Yes, we skip the following report. Now `redis-check-aof` tool is simple, and it doesn't support multiple mode at the same time, so if in `truncate` mode, i don't want to print extra info if there is no problem in AOF, just do `truncate` job without reporting analysis.

Reply:
I don't want to make redis-check-aof tool complex in core redis repo, actually there still are some other jobs which an AOF tool can do, such as grep -v "annotations", truncate AOF from a specific time, maybe support multi jobs together.